### PR TITLE
Asset file for trimmed captures

### PR DESCRIPTION
**Pull request by panos-lunarg**

### CODE REVIEW

**Review by w-pearson — COMMENTED** [review]
Some notes I have. I haven't checked if any of them were affected by further changes on #1644.

File: framework/decode/file_processor.cpp
Comment (panos-lunarg) [reply framework/decode/file_processor.cpp:1919]: Good catch. I'll try to apply this

File: framework/decode/file_processor.h
Comment (panos-lunarg) [reply framework/decode/file_processor.h:121]: Yes, I have in mind fixing this. When applying optimize the reported numbers are wrong so I have in mind investigating.

File: framework/encode/vulkan_state_writer.cpp
Comment (panos-lunarg) [reply framework/encode/vulkan_state_writer.cpp:1021]: The wrapper can be not-dirty the next time the assets are dumped (the next time `WriteDescriptorSetStateWithAssetFile` is called) in the same run.

However I think the way I handle things here is a bit problematic because the `WriteFunctionCall` in line 1024 can write a single `vkAllocateDescriptorSets` for multiple descriptor sets and the second time the assets are dumped some can be dirty and other not-dirty.
I need to handle these cases properly.

File: framework/encode/vulkan_state_writer.cpp
Comment (panos-lunarg) [reply framework/encode/vulkan_state_writer.cpp:1020]: Right. We can apply this optimization

File: framework/decode/file_processor.cpp
Comment (bradgrantham-lunarg) [inline framework/decode/file_processor.cpp:99]: Can you use `gfxrecon::util::filepath::GetBasedir` for this?

File: framework/decode/file_processor.cpp
Comment (bradgrantham-lunarg) [inline framework/decode/file_processor.cpp:111]: Can you use `gfxrecon::util::filepath::Join` for this?  I don't think you'd even need to check `absoluate_path_.empty()`.

File: framework/encode/api_capture_manager.h
Comment (bradgrantham-lunarg) [inline framework/encode/api_capture_manager.h:130]: Is this deleted in this PR just for cleanup?

File: framework/encode/capture_manager.cpp
Comment (bradgrantham-lunarg) [inline framework/encode/capture_manager.cpp:706]: I struggled to understand this logic.  I expected `write_assets_` to be set from `write_assets` and for `previous_write_assets_` to be the previous value of `write_assets_`.  I don't have a good suggestion except it looks like `write_assets_` is a one-shot, so could `write_assets` be `write_assets_requested` and `previous_write_assets_` be `write_assets_already_requested_`?  (Or some such)

File: framework/encode/capture_manager.cpp
Comment (bradgrantham-lunarg) [inline framework/encode/capture_manager.cpp:1086]: Please see if the functions `GetFilenameStem` and `GetFilenameExtension` in `gfxrecon::util::filepath` can be used for parsing out the extension.

File: framework/encode/api_capture_manager.h
Comment (panos-lunarg) [reply framework/encode/api_capture_manager.h:130]: More or less yes. Since now there are 2 runtime android options `GetRuntimeTriggerState` didn't serve any great purpose returning just the trigger state. Now `GetRuntimeTriggerEnabled`,  `GetRuntimeTriggerDisabled` and the new `RuntimeWriteAssetsEnabled` and `RuntimeWriteAssetsDisabled` fetch the runtime values themselves

File: framework/encode/capture_manager.cpp
Comment (bradgrantham-lunarg) [reply framework/encode/capture_manager.cpp:699]: I think I understand the thinking but the names didn't help explain what's going on in this block to me.  I think this is a confusing block of code with the current names but since it works and is a high-priority feature, lacking a better suggestion I think I'd like to get it in and come back to it.

File: framework/encode/vulkan_state_writer.cpp
Comment (bradgrantham-lunarg) [inline framework/encode/vulkan_state_writer.cpp:226]: Should this and the other instance of timing data written to `GFXRECON_WRITE_CONSOLE` use `GFXRECON_LOG_INFO` instead?

**Review by bradgrantham-lunarg — APPROVED** [review]
This is a good PR and adds a great feature!  Thank you.  I know you've been waiting a while for a review.

I think the following is still left before merging:
1. Probably should address use of WriteToFile in new WriteFileHeader but that can be in an issue for after this merge
1. I brought up plumbing the asset file filename through writestate - if desired that can be deferred to after the PR merge
1. GFXRECON_WRITE_CONSOLE sounds like left over debug prints so I think those should be either GFXRECON_LOG_DEBUG or at the very least INFO if you think it's still generally useful.
1. Have you addressed @w-pearson 's feedback to his and your satisfaction?
1. It looks like there are conflicts with current `dev`.  I think you can resolve those faster than I can but if you need to leave that to me I can try to resolve during Pacific time working hours.

File: framework/encode/vulkan_state_writer.cpp
Comment (w-pearson) [reply framework/encode/vulkan_state_writer.cpp:1025]: Was the single `vkAllocateDescriptorSets` call for multiple descriptor sets issue fixed?

**Review by w-pearson — COMMENTED** [review]
Looks like all of my initial feedback was addressed, though I did spot a few small things.

I'm not sure if I'll have time to do another full pass over this, so don't treat a lack of a review from me as a blocker. If I do have time I'll take a look though.

File: framework/decode/preload_file_processor.cpp
Comment (panos-lunarg) [reply framework/decode/preload_file_processor.cpp:341]: Good catch. The comments are valid code again.

File: framework/encode/vulkan_state_writer.cpp
Comment (panos-lunarg) [reply framework/encode/vulkan_state_writer.cpp:1029]: Yes. After some more thought I realized that this is handled correctly after all.

File: framework/encode/vulkan_state_writer.cpp
Comment (panos-lunarg) [reply framework/encode/vulkan_state_writer.cpp:230]: Turned these into `GFXRECON_LOG_INFO`

**Comment by panos-lunarg** [comment]
@bradgrantham-lunarg 
1. Good point. Will create a new issue for that.
2. I cherry-picked your commit
3. Were demoted to infos.
5. https://github.com/LunarG/gfxreconstruct/commit/dfa5b3e67838ad008e0c4b13e04b98c65c6e32cb caused a lot of conflicts. I hope I addressed all of them properly.

**Pull request merged**
